CRM/Core/Payment/PayJunction.php

```diff
@@ -57,6 +57,7 @@ public function __construct($mode, &$paymentProcessor) {
    *   the result in an nice formatted array (or an error object)
    */
   public function doDirectPayment(&$params) {
+    require_once 'PayJunction/pjClasses.php'; // necessary for event registrations 
```

Comment:
There is a similar pr https://github.com/civicrm/civicrm-core/pull/8300 done for eWay